src/Libraries/Microsoft.Extensions.AI/SpeechToText/LoggingSpeechToTextClient.cs

```diff
@@ -0,0 +1,199 @@
+    {
+        if (_logger.IsEnabled(LogLevel.Debug))
+        {
+            if (_logger.IsEnabled(LogLevel.Trace))
+            {
+                LogInvokedSensitive(nameof(GetStreamingTextAsync), "[audio speech stream]", AsJson(options), AsJson(this.GetService<SpeechToTextClientMetadata>()));
+            }
+            else
+            {
+                LogInvoked(nameof(GetStreamingTextAsync));
+            }
+        }
+
+        IAsyncEnumerator<SpeechToTextResponseUpdate> e;
+        try
+        {
+            e = base.GetStreamingTextAsync(audioSpeechStream, options, cancellationToken).GetAsyncEnumerator(cancellationToken);
+        }
+        catch (OperationCanceledException)
+        {
+            LogInvocationCanceled(nameof(GetStreamingTextAsync));
+            throw;
+        }
+        catch (Exception ex)
+        {
+            LogInvocationFailed(nameof(GetStreamingTextAsync), ex);
+            throw;
+        }
+
+        try
+        {
+            SpeechToTextResponseUpdate? update = null;
+            while (true)
+            {
+                try
+                {
+                    if (!await e.MoveNextAsync().ConfigureAwait(false))
+                    {
+                        break;
+                    }
+
+                    update = e.Current;
+                }
+                catch (OperationCanceledException)
+                {
+                    LogInvocationCanceled(nameof(GetStreamingTextAsync));
+                    throw;
+                }
+                catch (Exception ex)
+                {
+                    LogInvocationFailed(nameof(GetStreamingTextAsync), ex);
+                    throw;
+                }
+
+                if (_logger.IsEnabled(LogLevel.Debug))
+                {
+                    if (_logger.IsEnabled(LogLevel.Trace))
+                    {
+                        LogStreamingUpdateSensitive(AsJson(update));
+                    }
+                    else
+                    {
+                        LogStreamingUpdate();
+                    }
+                }
+
+                yield return update;
+            }
+
+            LogCompleted(nameof(GetStreamingTextAsync));
+        }
+        finally
+        {
+            await e.DisposeAsync().ConfigureAwait(false);
+        }
+    }
+
+    private string AsJson<T>(T value) => LoggingHelpers.AsJson(value, _jsonSerializerOptions);
+
+    [LoggerMessage(LogLevel.Debug, "{MethodName} invoked.")]
+    private partial void LogInvoked(string methodName);
+
+    [LoggerMessage(LogLevel.Trace, $"{{MethodName}} invoked: Audio contents: {{AudioContents}}. Options: {{{nameof(AI.SpeechToTextOptions)}}}. Metadata: {{{nameof(AI.SpeechToTextClientMetadata)}}}.")]
+    private partial void LogInvokedSensitive(string methodName, string audioContents, string SpeechToTextOptions, string SpeechToTextClientMetadata);
```

Comment:
These parameter names should be camelCased.

And the [LoggerMessage] message should not be using `nameof` to try to simulate the parameter names.

Reply:
@stephentoub I found this to be handy when renaming the methods as this happened frequently in this PR. Any reason in specific for avoiding it?

Reply:
How does it help with renames? If you rename AI.SpeechToTextOptions, for example, yes, the attribute will change... and then it'll be wrong, because it'll no longer map to the parameter name.

Reply:
I don't see how it helps at all. The placeholders have no associations with these types or methods and yet are referring to them in hopes that they remain named the same as the parameters. It can only hurt.